File: commands/bug.py
#!/usr/bin/env python3
import os
import discord
from github import Github


async def bug(ctx, title, description):

    embed = discord.Embed(color=0xFEE9B6, title="⏳  Loading...")
    interaction = await ctx.respond(embed=embed)

    G = Github(os.getenv("GITHUB_TOKEN"))
    repo = G.get_repo("NicPWNs/MEGABOT")
    issue = repo.create_issue(title=title, body=description)

    embed = discord.Embed(
        color=0x69A24A,
        title=f"🪲  Created New Bug #{issue.number}",
        description=f"[{issue.title}](https://github.com/NicPWNs/MEGABOT/issues/{issue.number})",
    )

    await interaction.edit_original_response(embed=embed)

File: commands/bug.ts
import {
  DiscordInteraction,
  InteractionResponse,
  Colors,
} from "../types/discord";
import { embedResponse, errorResponse } from "../utils/discord";
import { getOptionValue, getUser } from "../handlers/commands";
import { Octokit } from "@octokit/rest";

export async function handleBug(
  interaction: DiscordInteraction,
): Promise<InteractionResponse> {
  const title = getOptionValue<string>(interaction, "title")!;
  const description = getOptionValue<string>(interaction, "description")!;
  const user = getUser(interaction);

  const githubToken = process.env.GITHUB_TOKEN;
  if (!githubToken) {
    return errorResponse("GitHub integration not configured.");
  }

  try {
    const octokit = new Octokit({ auth: githubToken });

    const body = `${description}\n\n> Submitted by \`${user.username}\``;

    const { data: issue } = await octokit.issues.create({
      owner: "NicPWNs",
      repo: "MEGABOT",
      title,
      body,
      labels: ["bug"],
      assignees: ["NicPWNs"],
    });

    return embedResponse({
      title: `🪲 Created New Bug #${issue.number}`,
      description: `[${issue.title}](${issue.html_url})`,
      color: Colors.GREEN,
    });
  } catch (error) {
    console.error("GitHub API error:", error);
    return errorResponse(
      "Failed to create bug report. Please try again later.",
    );
  }
}
